vsintegration/src/FSharp.Editor/Hints/InlineParameterNameHints.fs

```diff
@@ -76,49 +74,51 @@ module InlineParameterNameHints =
             false
 
     let isUnionCaseValidForHint (symbol: FSharpUnionCase) (symbolUse: FSharpSymbolUse) =
-        symbolUse.IsFromUse && symbol.DisplayName <> "(::)"
+        symbolUse.IsFromUse
+        && symbol.DisplayName <> "(::)"
+        // If a case does not use field names, don't even bother getting applied argument ranges
+        && Seq.toList symbol.Fields |> Seq.exists doesFieldNameExist
 
-    let getHintsForMemberOrFunctionOrValue
+    member _.getHintsForMemberOrFunctionOrValue
         (sourceText: SourceText)
-        (parseResults: FSharpParseFileResults)
         (symbol: FSharpMemberOrFunctionOrValue)
         (symbolUse: FSharpSymbolUse)
         =
 
-        let parameters = symbol.CurriedParameterGroups |> Seq.concat
-        let argumentLocations = parseResults |> getArgumentLocations symbolUse
-
-        let tupleRanges = argumentLocations |> getTupleRanges
-        let curryRanges = parseResults |> getCurryRanges symbolUse
+        if isMemberOrFunctionOrValueValidForHint symbol symbolUse then
+            let parameters = symbol.CurriedParameterGroups |> Seq.concat
+            let argumentLocations = getArgumentLocations symbolUse
 
-        let ranges =
-            if tupleRanges |> (not << Seq.isEmpty) then
-                tupleRanges
-            else
-                curryRanges
-            |> Seq.filter (fun range -> argumentLocations |> (not << isNamedArgument range))
+            let tupleRanges = argumentLocations |> getTupleRanges
+            let curryRanges = getCurryRanges symbolUse
 
-        let argumentNames = ranges |> Seq.map (getSourceTextAtRange sourceText)
-
-        parameters
-        |> Seq.zip ranges // Seq.zip is important as List.zip requires equal lengths
-        |> Seq.where (snd >> doesParameterNameExist)
-        |> Seq.zip argumentNames
-        |> Seq.choose (fun (argumentName, (range, parameter)) ->
-            if argumentName <> parameter.DisplayName then
-                Some(getParameterHint (range, parameter))
-            else
-                None)
-        |> Seq.toList
+            let ranges =
+                if tupleRanges |> (not << Seq.isEmpty) then
+                    tupleRanges
+                else
+                    curryRanges
```

Comment:
Can this just be inverted, so we dont have piping into composition?
```suggestion
                if Seq.isEmpty tupleRanges then
                    curryRanges
                else
                    tupleRanges
```

Reply:
Yep, good point